InteractiveHtmlBom/web/render.js

```diff
@@ -632,6 +640,28 @@ function pointWithinDistanceToSegment(x, y, x1, y1, x2, y2, d) {
   return dx * dx + dy * dy <= d * d;
 }
 
+function modulo(n, mod) {
+  return ((n % mod) + mod ) % mod;
+}
+
+function pointWithinDistanceToArc(x, y, xc, yc, radius, startangle, endangle, d) {
+  var angle1 = modulo(startangle, 360.0);
+  var angle2 = modulo(endangle, 360.0);
+  var dx = x - xc;
+  var dy = y - yc;
+  var r = Math.sqrt(dx * dx + dy * dy);
```

Comment:
Avoid expensive sqrt operation by comparing squares of distances.

Reply:
I sketched out the geometry a couple ways and kept coming back to needing at least one square root.  If you've got a clever way that avoids a sqrt, let me know.

`(r-radius)^2` != `|r^2 - radius^2|`

Reply:
I think I figured it out, stand by...